internal/manifest/v1beta1/spec_resolver.go

```diff
@@ -174,27 +172,28 @@ func (m *ManifestSpecResolver) getValuesFromConfig(
 }
 
 var (
-	ErrChartConfigObjectInvalid = errors.New("chart config object of .spec.config is invalid")
-	ErrConfigObjectInvalid      = errors.New(".spec.config is invalid")
+	ErrConfigObjectInvalid = errors.New(".spec.config is invalid")
 )
 
-func parseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
+func ParseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
 	var configs []interface{}
-	installConfigObj, decodeOk := decodedConfig.(map[string]interface{})
-	if !decodeOk {
+
+	type ConfigsYaml struct {
+		Configs []interface{}
+	}
+
+	var configsValues ConfigsYaml
+	err := mapstructure.Decode(decodedConfig, &configsValues)
```

Comment:
I think we don't need to introduce this mapstructure library just for this purpose, when we introduce some new library, please always keep in mind it introduce dependencies to 3rd parties which is not under our control, especially more things for security scanner to bother us.

Regarding this issue, I think we can just return an empty `config` if the cast decoding not return `decodeOk`, right? or do you have specific concern here?

Reply:
No I was just using it for the decoding.